### ISSUE TITLE
Old code with language files and MySQL

### ISSUE THREAD

**Issue by amnuts** [amnuts/Amnuts#6]
I found a copy of version 2.2.2 yesterday where I had added a hash table for lookups on language files and started to integrate MySQL for storage of messages and profiles.

Putting it here as a reminder to one day resurrect that code and properly look into it.

Pretty sure I have some test code somewhere that used sqlite instead, and I'd prefer that over MySQL.  So try to find it!

**Comment by Uzume** [comment]
Yes, but MySQL would allow you to run multiple talkers (perhaps on different hosts) using the same database! I know that is definitely a priority for you (wink wink). Who needs netlinks when one can just use a single shared database (with shared login sessions and credentials, etc).

I think I would lean toward a non-relational semi-structured database when it comes to use in talkers (but query language choice then becomes a key issue).

As far as low footprint open source SQL alternatives go, I would consider the following:
- [TiDB](https://github.com/pingcap/tidb) (supposedly MySQL compatible even) written in Go and Rust (TiKV) with Apache 2.0 license
- [Firebird](https://github.com/FirebirdSQL/firebird) (forked from Borland InterBase) written in C++ and C with Mozilla 1.1 style license

There are also a few Java ones around (e.g., [H2](https://github.com/h2database/h2database)) but I would not recommend those unless you are writing something in a language that targets the Java/JVM platform or want to run a heavier weight server (e.g., [Apache Derby](https://github.com/apache/derby) aka Cloudscape).

**Comment by blindsight** [comment]
It would be nice to have a data layer and just write drivers. MySQL/MariaDb, TiDB, Firebird, MongoDB and such would just be a choice by the person setting it up. I've been tempted to run a talker again in a container. A bunch of security holes for these days.

**Comment by amnuts** [comment]
> I've been tempted to run a talker again in a container. A bunch of security holes for these days.

I know it's been half a year (and in terms of old school talkers, that's like yesterday), but I added a bit of a container setup to Amnuts the other day.  Kind of a work in progress, but if you fancy seeing if it's practical for use or needs: https://github.com/amnuts/Amnuts/pull/12

**Comment by Uzume** [comment]
Instead of putting this issue here--why not post the code here as a branch?

**Comment by amnuts** [comment]
Well, as that was 11 years ago, I'd have to try to find it again.  And even if I could find it, I'd have to try to bring it up-to-date with the latest code.  Would probably just be easier to start from scratch on it... And if I were to do that, I'd go for SQLite instead of MySQL - more portable and definitely vibes (not in the AI sense) more with Amnuts.
